### PR TITLE
25 continue server

### PR DESCRIPTION
**Pull request by furaiev**

### CODE REVIEW

File: src/pages/Authorized/Authorized.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { Route, Switch, Redirect } from 'react-router-dom';
+import { Tasklist } from '../../Pages/Tasklist';
+import { Main } from '../../partials/Main';
+import { NotFound } from '../../Pages/NotFound';
+import { Task } from '../../Pages/Task';
+import { User } from '../../components/User';
+
+export class Authorized extends Component {
+  render() {
```
[the remark this answers is not in the document]
Comment (furaiev) [reply src/pages/Authorized/Authorized.js]: I order to get props and pass them further

**Comment by furaiev** [comment]
@iliatcymbal updated.
Dummy components with props are really cool) Thanks!

File: src/components/TasksSummary/TasksSummary.js [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
 
+  componentDidMount() {
+    getInfo().then(info => this.setState(info));
+  }
+
   render() {
     return (
       <div id="tasksSummary">
-        <li>Hello, {this.state.userName || 'user'}</li>
+        <li>Hello, {this.props.login ? this.props.login.firstName : 'friend'}</li>
```
Comment (iliatcymbal) [inline src/components/TasksSummary/TasksSummary.js]: use defaultProps for default props if you need it

File: src/pages/Registration/Registration.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        window.alert('Cannot login');
+        console.log('Cannot login', err);
+        this.setState({ loading: false });
+      });
+  };
+
+  render() {
+    const { loading } = this.state;
+    return (
+      loading ? <Loader /> : <Form onSubmit={form => this.submit(form)} />
```
Comment (iliatcymbal) [inline src/pages/Registration/Registration.js]: do not create new function:
```
onSubmit={this.submi}
```

File: src/pages/Tasklist/Tasklist.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    if (key === 'delete') {
+      deleteTask(task.id)
+        .then((data) => {
+          const tasks = todayTasks.filter(task => task.id !== data.id);
+          tasksInWeek[day] = tasks;
+          this.setState({ tasksInWeek });
+        });
+    }
+  };
```
Comment (iliatcymbal) [inline src/pages/Tasklist/Tasklist.js]: I would separate delete and update stuff (two different methods)

File: src/components/TasksSummary/TasksSummary.js (hunk 2 above)
Comment (furaiev) [reply src/components/TasksSummary/TasksSummary.js]: will be updated in the next PR

File: src/pages/Registration/Registration.js (hunk 3 above)
Comment (furaiev) [reply src/pages/Registration/Registration.js]: will be updated in the next PR

File: src/pages/Tasklist/Tasklist.js (hunk 4 above)
Comment (furaiev) [reply src/pages/Tasklist/Tasklist.js]: will be updated in the next PR

**Pull request merged**